packages/react-text/Spec.md

```diff
@@ -0,0 +1,150 @@
+We're also introducing a new concept of wrappers for the main design variants (i.e. 'Title', 'Subtitle', 'Caption') to improve readability and semantics of the code. The wrappers are expected to have fixed styles in terms of size and font family, but flexible for the other supported props in Text.
+These wrappers follow the Fluent UI language so for any deviation, regarding the fixed styles mentioned above, should use the Text component instead, given that this is fully customizable.
+
+| Property      | Type                                                                   | Default value | Comments |
+| ------------- | ---------------------------------------------------------------------- | ------------- | -------- |
+| as            | "span" \| "p" \| "h1" \| "h2" \| "h3" \| "h4" \| "h5" \| "h6" \| "pre" | "span"        |          |
+| wrap          | boolean                                                                | true          |          |
+| truncate      | boolean                                                                | false         |          |
+| block         | boolean                                                                | false         |          |
+| italic        | boolean                                                                | false         |          |
+| underline     | boolean                                                                | false         |          |
+| strikethrough | boolean                                                                | false         |          |
+| size          | number - 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000             | 300           |          |
+| font          | string - "base" \| "monospace" \| "numeric"                            | "base"        |          |
+| weight        | string - "regular" \| "medium" \| "semibold"                           | "regular"     |          |
+| align         | string - "start" \| "center" \| "end" \| "justify"                     | "start"       |          |
+
+### make-styles rules
+
+Finally, we're also making available the styling used internally in Text for a lightweight styling solution if you want to avoid using Text or the added JavaScript layers of Text/wrappers.
+This is achieved with the `make-styles` rules being available to the user so that they can follow the Fluent design standards for Typography.
+
+## Sample Code
+
+### Using Text
+
+```jsx
+<Text weight="semibold" size={1000}>This text is semibold and huge.</Text>
+
+<Text align="end" italic>Text aligned to the end</Text>
+
+<Text strikethrough>This text has a strikethrough.</Text>
+```
+
+### Using the wrappers
+
+```jsx
+<Display bold>This text is huge.</Display>
+<LargeTitle>This is a large title.</LargeTitle>
+<Title as="h1">Title</Title>
+<Subtitle as="h2">Sub title</Subtitle>
+<Caption>Captioned</Caption>
+```
+
+### Using styles directly
+
+```jsx
+import { useTypographyStyles } from '@fluentui/react-text';
```

Comment:
Could you provide more examples how this would work for the different variants of `Text` ? would there be multiple hooks such as:

* useSubtitleStyles
* useTitleStyles
* ...

for example our current style hook pattern leverages the component state:

```typescript
useTypographyStyles(state);
```

Reply:
There'll only be one hook to return all the typography styles. `useTypographyStyles` will return `title`, `subtitle`, `display`, etc.
We can add more examples to make that clear.

Regarding state, what would be the advantage for this? Wouldn't this force the user to make the component's state type extend the hook?

Reply:
yes, it would force the user to create component variants with the style hook of the original `Text` (and variants) style hook, which AFAIK is why we went with our composition architecture

Reply:
Text styles are composed manually as defined in the spec, this hook will be used by the wrappers and external consumers.
The concept of variants does not exist for Text like it did in Fabric.

Reply:
does that mean composition won't be used with the `Text` component ? we have assumed that our consumers can (and should) use this pattern:

```typescript
let state = useTextState(props);
state = useTextStyles(state);
return renderText(state);
```

since the `Text` props have all the props to compose the styles it should be possible for us internally even to use the style hook exposed by the `Text` component, so we use the same API that our consumers will be using.

Reply:
Again, `useTypographyStyles` and `useTextStyles` have different purposes.
- `useTypographyStyles` - Returns the different variant styles for composing a Title, Subtitle, Display, etc
- `useTextStyles` - Updates state's className to apply the Text styles (weight, size, italics, strikethrough, etc)
As discussed before, Typography hook is strictly to provide a lightweight alternative to using the wrappers we will implement, which is why it's detached from the Text hook and does not need any state.

Reply:
Alright, let's shelf this discussion for now, we agree that text styles should reused let's see how that will work in practice :)